### DISCUSSION TITLE
Include file(s) that are excluded with .gitignore

### DISCUSSION THREAD

**Discussion by blinxen** [BurntSushi/ripgrep#2512] [Q&A] [answered]
I read the documentation and searched for this in some forums but I can't find a solution.

I have the following scenario:

When I am working on a project I like to keep a file called `TODO` around to document what I am doing currently. I create this file in my nextcloud directory (to have it in the cloud) and symlink it to the project repository. Because this file consists of my personal notes, I don't want it to be included in `git` so I ignore it in my global gitignore (`$HOME/.gitignore`). `ripgrep`s automatic filtering ignores this file when grepping (which is OK for some cases but not for this one). I would like to keep the `TODO` file ignored in `git` but I want `ripgrep` to not ignore it when grepping. Is there a way to achieve this? Obviously I don't want to change my `gitignore` files.

My first solution was the following `ripgrep` config file (which contains similiar cases like the `TODO` file):

```
# Include hidden files and directories
--hidden
# Include symlinks
--follow
# Include docker-compose files
--glob=docker-compose*
# Include .env files
--glob=.env
# Include TODO files
--glob=TODO
```

This does not work because of my misunderstanding of `--glob` but I would like to have this kind of solution.

Thanks for the help!

**Comment by VladimirMarkelov** [comment, marked as answer]
Have you tried using `.rgignore`? See: https://github.com/BurntSushi/ripgrep/discussions/2436

**Reply by blinxen** [reply]
Ah, I somehow did not see `.rgignore` in [GUIDE.md](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering). This is exactly what I was looking for. Thanks! Now I have to find out how to make this file global instead of adding it to all directories.

**Reply by BurntSushi** [reply]
You want the `--ignore-file` flag perhaps. It might not work here because it has low precedence. I can't remember which one wins: a global gitignore or an `--ignore-file`.
